### PR TITLE
add support for async function for ws callback

### PR DESCRIPTION
**Pull request by jkh911208**
this add support for passing async function for subscription

### CODE REVIEW

**Comment by jkh911208** [comment]
@bretterer can you review this?

File: src/rivian/ws_monitor.py [hunk 1]
```diff
@@ -151,7 +152,10 @@ async def _receiver(self) -> None:
                         self._connection_ack.set()
                     elif data_type == "next":
                         if (_id := data.get("id")) in self._subscriptions:
-                            self._subscriptions[_id][0](data)
+                            if inspect.iscoroutinefunction(self._subscriptions[_id][0]):
```
Comment (natekspencer) [inline src/rivian/ws_monitor.py:155]: Please change the type of `_subscriptions` to allow for an Awaitable. You'll also have to adjust the typing anywhere that modifies `_subscriptions` to make sure it is appropriate as well.

**Review by natekspencer — CHANGES_REQUESTED** [review]

File: src/rivian/ws_monitor.py (hunk 1 above)
Comment (jkh911208) [reply src/rivian/ws_monitor.py:155]: @natekspencer i don't think it is needed since async function is also Callable type same as normal function.

Comment (natekspencer) [reply src/rivian/ws_monitor.py:155]: Oh actually, yeah you're probably right. I need more coffee 😅

Comment (natekspencer) [reply src/rivian/ws_monitor.py:155]: It looks like mypy doesn't like it still though. But maybe you just need to do something like `_fn = self._subscriptions[_id][0]` and then the logic you added

Comment (jkh911208) [reply src/rivian/ws_monitor.py]: @natekspencer Fixed, it is passing mypy now

**Review by natekspencer — APPROVED** [review]

**Pull request merged**